src/instruction.rs

```diff
@@ -774,20 +780,35 @@ impl fmt::Display for Instruction {
             Instruction::GateDefinition(GateDefinition {
                 name,
                 parameters,
-                matrix,
+                specification,
                 r#type,
             }) => {
                 let parameter_str: String = parameters.iter().map(|p| p.to_string()).collect();
                 writeln!(f, "DEFGATE {}{} AS {}:", name, parameter_str, r#type)?;
-                for row in matrix {
-                    writeln!(
-                        f,
-                        "\t{}",
-                        row.iter()
-                            .map(|cell| format!("{}", cell))
-                            .collect::<Vec<String>>()
-                            .join(",")
-                    )?;
+                match specification {
+                    GateSpecification::Matrix(matrix) => {
+                        for row in matrix {
+                            writeln!(
+                                f,
+                                "\t{}",
+                                row.iter()
+                                    .map(|cell| format!("{}", cell))
```

Comment:
nit: given that `format!("{}", cell)` and `cell.to_string()` are the same, we should stick to one or the other.

Reply:
I'd prefer `to_string()` for converting a single item to string and leave `format!()` for all other cases (building strings from multiple values, etc.)